NPE when sorting files

## Issue explanation (write below this line)



## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager
* __Version:__ 3.6.7
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux Android 10 - 29
* __Device:__ mdh50lm
* __Model:__ LM-Q730
* __Product:__ mdh50lm
*Crash log*



```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:399)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void java.util.List.sort(java.util.Comparator)' on a null object reference
at java.util.Collections.sort(Collections.java:206)
at com.amaze.filemanager.ui.fragments.MainFragment$4.doInBackground(MainFragment.java:1472)
at com.amaze.filemanager.ui.fragments.MainFragment$4.doInBackground(MainFragment.java:1469)
at android.os.AsyncTask$3.call(AsyncTask.java:378)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 3 more

```


----------------------------------------


-- 
N Clark

This email and its attachments may contain information that is confidential and/or protected by the attorney-client privilege. If you are not the intended recipient of this email, or believe you have received it in error, please notify the sender and promptly delete or destroy this email, all attachments, and any copies you may have.

> ## Issue explanation (write below this line)
> 
> 
> 
> ## Exception
> * __App Name:__ Amaze File Manager
> * __Package:__ com.amaze.filemanager
> * __Version:__ 3.6.7
> * __User Action:__ UI Error
> * __Request:__ Application crash
> * __OS:__ Linux Android 10 - 29
> * __Device:__ mdh50lm
> * __Model:__ LM-Q730
> * __Product:__ mdh50lm
> *Crash log*
> 
> 
> 
> ```
> java.lang.RuntimeException: An error occurred while executing doInBackground()
> at android.os.AsyncTask$4.done(AsyncTask.java:399)
> at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
> at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
> at java.util.concurrent.FutureTask.run(FutureTask.java:271)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
> at java.lang.Thread.run(Thread.java:919)
> Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void java.util.List.sort(java.util.Comparator)' on a null object reference
> at java.util.Collections.sort(Collections.java:206)
> at com.amaze.filemanager.ui.fragments.MainFragment$4.doInBackground(MainFragment.java:1472)
> at com.amaze.filemanager.ui.fragments.MainFragment$4.doInBackground(MainFragment.java:1469)
> at android.os.AsyncTask$3.call(AsyncTask.java:378)
> at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> ... 3 more
> 
> ```
> 
> 
> ----------------------------------------
> 
> 
> -- 
> N Clark
> 
> This email and its attachments may contain information that is confidential and/or protected by the attorney-client privilege. If you are not the intended recipient of this email, or believe you have received it in error, please notify the sender and promptly delete or destroy this email, all attachments, and any copies you may have.
> 

-

--- a/app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java
+++ b/app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java
@@ -1465,9 +1465,10 @@ public class MainFragment extends Fragment
   }
 
   public void onSearchCompleted(final String query) {
-    final ArrayList<LayoutElementParcelable> elements = mainFragmentViewModel.getListElements();
+    final @Nullable ArrayList<LayoutElementParcelable> elements = mainFragmentViewModel.getListElements();
     if (elements == null) {
       Log.e(TAG, "No search results, cannot sort!");
+      return;
     }
     if (!mainFragmentViewModel.getResults()) {
       // no results were found

